sources/web/datalab/jupyter.ts

```diff
@@ -64,14 +66,19 @@ var templates: common.Map<string> = {
  */
 var appSettings: common.Settings;
 
-function pipeOutput(stream: NodeJS.ReadableStream, port: number, error: boolean) {
+function pipeOutput(server: JupyterServer, stream: NodeJS.ReadableStream, port: number, error: boolean) {
   stream.setEncoding('utf8');
+  var listeningMessage = `The Jupyter Notebook is running at: http://127.0.0.1:${port}`;
+
   stream.on('data', (data: string) => {
     // Jupyter generates a polling kernel message once every 3 seconds
     // per kernel! This adds too much noise into the log, so avoid
     // logging it.
 
     if (data.indexOf('Polling kernel') < 0) {
+      if (data.indexOf(listeningMessage) >= 0) {
```

Comment:
No, I don't think this can be removed. This is arguably the most important safeguard that is being added.

Specifically, this check ensures that we do not send any traffic to the proxy until after we confirm that the proxy is pointing at the Jupyter server we think it is.